src/components/OpenWith/OpenWith.js

```diff
@@ -86,16 +100,38 @@ class OpenWith extends PureComponent<Props> {
     }
 
     /**
-     * Fetches the list of potential integrations on load
      *
      * @private
      * @inheritdoc
      * @return {void}
      */
     componentDidMount() {
-        /* no-op */
+        const { fileId }: Props = this.props;
```

Comment:
Maybe I am just not seeing it, but you will need to put this logic in componentDidUpdate if the fileId changes

Reply:
I will add this when I get to actual rendering in my next PR.